### PR TITLE
update to use latest kzg lib

### PR DESCRIPTION
**Pull request by roberto-bayardo**

### CODE REVIEW

File: beacon-chain/core/blob/sidecar.go [hunk 1]
```diff
@@ -130,7 +130,7 @@ func ValidateBlobsSidecar(slot types.Slot, root [32]byte, commitments [][]byte,
 	var y bls.Fr
 	EvaluatePolyInEvaluationForm(&y, aggregatedPoly, &x)
 
-	b, err := verifyKZGProof(aggregatedPolyCommitment, &x, &y, sidecar.AggregatedProof)
+	b, err := kzg.VerifyKZGProof(*(*[48]byte)(aggregatedPolyCommitment), &x, &y, *(*[48]byte)(sidecar.AggregatedProof))
```
Comment (roberto-bayardo) [inline beacon-chain/core/blob/sidecar.go]: figured its safe enough to use these slice->array conversions without length checks, though they would panic if the underlying slice happens to be shorter than expected.

Comment (Inphi) [reply beacon-chain/core/blob/sidecar.go]: Safer to use the [ToBytes48](https://github.com/Inphi/prysm/blob/eip-4844/encoding/bytesutil/bytes.go#L114) helper function as that truncates the length if the slice is shorter than expected.

Comment (roberto-bayardo) [reply beacon-chain/core/blob/sidecar.go]: ah nice, changed all type conversions to use those utilities.

**Review by Inphi — APPROVED** [review]

**Comment by terencechain** [comment]
Thanks @roberto-bayardo I'll also use this for https://github.com/prysmaticlabs/prysm/pull/11621 🙏🏼

**Comment by roberto-bayardo** [comment]
@terencechain great to hear! However: I am going to have more kzg related changes incoming, as I get ready to move most of the kzg stuff still in the clients into go-kzg. it might make sense to wait until I'm closer to done. I'm doing it incrementally within devnet mostly to make sure I don't break anything.

**Pull request merged**